### PR TITLE
licensing: do not hardcode button length

### PR DESCRIPTION
**Pull request by ABBAPOH**
As seen in the TypeScript based [dmg-license](https://github.com/argv-minus-one/dmg-license/blob/4268f2e822944fd670c1e197596396f233d6484e/src/Labels.ts#L284) implementation, the byte before the buttons should contain the length of the encoded string. Currently, it is hardcoded for the English buttons which is incorrect.

### CODE REVIEW

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
It's very difficult to judge whether a patch is necessary when the PR description contains no details about what problem the PR is trying to address, or how to test the output.

Superficially, this doesn't seem to make sense - the `join` you're performing won't produce the same output as the manually constructed code it is replacing.

**Comment by freakboy3742** [comment]
> > the join you're performing won't produce the same output as the manually constructed code it is replacing.
> 
> Can you elaborate? I printed the text and it was the same as before (for English version); unless I broke smth while minimising the code but it seems working (provided some other hacks, e.g. change the ID - I am not sure how to fix them properly for now)

Basic code inspection of the proposed change. The old code puts single byte markers between each element in `buttons`. The new code does not. Removing arbitrary bytes from a byte string is rarely a good move.

When you originally submitted this PR, it had literally no description, so it wasn't at all clear why this change was being proposed. You've updated the description to point at a TypeScript implementation; the mapping from that implementation to this one isn't at all obvious to me, and I can't say I'm especially motivated to spend a couple of hours getting up to speed on an alternate implementation in a language I don't spend a lot of time using.

In short - how do I validate that this PR is correct? Right now, the code works in my testing. I won't claim it's bug free, but if you want to propose a change, a key part of that process is to provide a demonstration of something that *doesn't* work  with the current code, but *does* work with the proposed change - or, at the very least, a reference to a specification that demonstrates the algorithm being used at present is incorrect.

**Comment by ABBAPOH** [comment]
> The old code puts single byte markers between each element in buttons. The new code does not

The first parameter of the to_bytes() function is the length of of the buffer which is 1.
`>>> int(30).to_bytes(1, 'big')
b'\x1e'
>>> int(255).to_bytes(1, 'big')
b'\xff'
>>> int(300).to_bytes(1, 'big')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: int too big to convert`

> In short - how do I validate that this PR is correct? 

Current implementation does not support additional licences at all. I have a subsequent patch that fixes [combobox](https://github.com/dmgbuild/dmgbuild/issues/56) as well as custom buttons. I'll upload a fix with some test files later, probably will do in this PR.

**Review by freakboy3742 — APPROVED** [review]
Ok - now I see what it's doing. 

I'm happy to merge this as is, on the basis that there really isn't a test suite to speak of at the moment. That's something I'd very much like to address, but it doesn't need to stand in the way of this fix.
